packages/core/android/src/main/java/com/segment/analytics/reactnative/core/RNAnalyticsModule.kt

```diff
@@ -39,11 +42,67 @@ class RNAnalyticsModule(context: ReactApplicationContext): ReactContextBaseJavaM
 
     companion object {
         private var singletonJsonConfig: String? = null
+        private var key: String? = null
+        private var versionKey = "version"
+        private var buildKey = "build"
+    }
+
+    private fun getPackageInfo(): PackageInfo {
+        val packageManager = reactApplicationContext.packageManager
+        try {
+            return packageManager.getPackageInfo(reactApplicationContext.packageName, 0)
+        } catch (e: PackageManager.NameNotFoundException) {
+            throw AssertionError("Package not found: " + reactApplicationContext.packageName)
+        }
+    }
+
+    /**
+     * Tracks application lifecycle events - Application Installed, Application Updated and Application Opened
+     * This is built to exactly mirror the application lifecycle tracking in analytics-android
+     */
+    private fun trackApplicationLifecycleEvents() {
+        // Get the current version.
+        var packageInfo = this.getPackageInfo()
+        val currentVersion = packageInfo.versionName
+        val currentBuild = packageInfo.versionCode
+
+        // Get the previous recorded version.
+        val sharedPreferences = getSegmentSharedPreferences(reactApplicationContext, key)
+        val previousVersion = sharedPreferences.getString(versionKey, null)
+        val previousBuild = sharedPreferences.getInt(buildKey, -1)
+
+        // Check and track Application Installed or Application Updated.
+        if (previousBuild == -1) {
+            var installedProperties = Properties()
+            installedProperties[versionKey] = currentVersion
+            installedProperties[buildKey] = currentBuild
+            analytics.track("Application Installed", installedProperties)
+        } else if (currentBuild != previousBuild) {
+            var updatedProperties = Properties()
+            updatedProperties[versionKey] = currentVersion
+            updatedProperties[buildKey] = currentBuild
+            updatedProperties["previous_$versionKey"] = previousVersion
+            updatedProperties["previous_$buildKey"] = previousBuild
+            analytics.track("Application Updated", updatedProperties)
+        }
+
+        // Track Application Opened.
+        var appOpenedProperties = Properties()
+        appOpenedProperties[versionKey] = currentVersion
+        appOpenedProperties[buildKey] = currentBuild
+        analytics.track("Application Opened", appOpenedProperties)
+
+        // Update the recorded version.
+        val editor = sharedPreferences.edit()
+        editor.putString(versionKey, currentVersion)
+        editor.putInt(buildKey, currentBuild)
+        editor.apply()
     }
 
     @ReactMethod
     fun setup(options: ReadableMap, promise: Promise) {
         val json = options.getString("json")
+        val writeKey = options.getString("writeKey")
```

Comment:
Is this used?

Reply:
Yes - look at line 117 and 70. It's used as a storage key for shared preferences.
Good spot though, I should have used the `writeKey` variable on line 117, but I forgot to replace it. Will go now!